python_modules/dagit/dagit/app.py

```diff
@@ -69,22 +69,28 @@ def info_view():
     )
 
 
-def notebook_view(request_args):
+def notebook_view(context, request_args):
+    context = check.inst_param(context, "context", IWorkspaceProcessContext)
+    repo_location_name = request_args["repoLocName"]
+    repo_name = request_args["repoName"]
     check.dict_param(request_args, "request_args")
 
-    # This currently provides open access to your file system - the very least we can
-    # do is limit it to notebook files until we create a more permanent solution.
     path = request_args["path"]
     if not path.endswith(".ipynb"):
         return "Invalid Path", 400
 
-    with open(os.path.abspath(path)) as f:
-        read_data = f.read()
-        notebook = nbformat.reads(read_data, as_version=4)
-        html_exporter = HTMLExporter()
-        html_exporter.template_file = "basic"
-        (body, resources) = html_exporter.from_notebook_node(notebook)
-        return "<style>" + resources["inlining"]["css"][0] + "</style>" + body, 200
+    # get ipynb content from grpc call
+    request_context = context.create_request_context()
+    res = request_context.get_external_notebook_data(repo_name, repo_location_name, path)
+    notebook_content = res.content
+    check.inst_param(notebook_content, "notebook_content", bytes)
+
+    # parse content to HTML
+    notebook = nbformat.reads(notebook_content, as_version=4)
```

Comment:
this is still scaring me a little bit security-wise. Are we confident that nbformat.reads doesn't do anything resembling code execution?

Reply:
I dont think so: https://github.com/jupyter/nbformat/blob/master/nbformat/reader.py#L39 - it seems only parse the raw json into html

Reply:
per offline convo, even though nbformat doesn't execute the notebook, there could still be bad js, so we decided to add an `include_notebook_route` param to `instantiate_app_with_views` and default it to false for disabling the route by default: https://github.com/dagster-io/dagster/pull/4515/commits/79d5e5f91711e296fe3782f189ecb4b3e03f55af 

reference: https://jupyter-notebook.readthedocs.io/en/stable/security.html#notebook-security